hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -2169,11 +2152,20 @@ public AbfsClient getClient() {
     return this.client;
   }
 
+  public AbfsClient getClient(AbfsServiceType serviceType) {
+    return this.clientHandler.getClient(serviceType);
+  }
+
   @VisibleForTesting
   void setClient(AbfsClient client) {
     this.client = client;
   }
 
+  @VisibleForTesting
+  void setClientHandler(AbfsClientHandler clientHandler) {
```

Comment:
lets not add another setter for test,  lets use the power of mock.

Mockito.doReturn(requiredClientHandler).when(client).getClientHandler().


But this would require to have getClientHandler calls instead of field call inside store.

Reply:
Lets do for getClient(serviceType) as well.

Reply:
Removed setter for client Handler.
Thanks for pointing this out